feature/experimental/policy/policy_vrf_selection/ate_tests/base_vrf_selection/base_vrf_selection_test.go

```diff
@@ -155,100 +167,103 @@ func configInterfaceDUT(i *oc.Interface, me, peer *attrs.Attributes, subintfinde
 }
 
 // Configure Network instance on the DUT
-func configNetworkInstance(name string, peer *attrs.Attributes) *oc.NetworkInstance {
+func configNetworkInstance(name string, intf *oc.Interface, id uint32) *oc.NetworkInstance {
 	d := &oc.Root{}
 	ni := d.GetOrCreateNetworkInstance(name)
 
 	ni.Type = oc.NetworkInstanceTypes_NETWORK_INSTANCE_TYPE_L3VRF
-	static := ni.GetOrCreateProtocol(oc.PolicyTypes_INSTALL_PROTOCOL_TYPE_STATIC, *deviations.StaticProtocolName)
-	ipv4Nh := static.GetOrCreateStatic("0.0.0.0/0").GetOrCreateNextHop("0")
-	ipv4Nh.NextHop, _ = ipv4Nh.To_NetworkInstance_Protocol_Static_NextHop_NextHop_Union(peer.IPv4)
-	ipv6Nh := static.GetOrCreateStatic("::/0").GetOrCreateNextHop("0")
-	ipv6Nh.NextHop, _ = ipv6Nh.To_NetworkInstance_Protocol_Static_NextHop_NextHop_Union(peer.IPv6)
-	ipv4Nh.Recurse = ygot.Bool(true)
-	ipv6Nh.Recurse = ygot.Bool(true)
+	niIntf := ni.GetOrCreateInterface(*intf.Name)
```

Comment:
Description of xpath /network-instances/network-instance/interfaces/interface/config/id  from:
`https://openconfig.net/projects/models/schemadocs/yangdoc/openconfig-network-instance.html#network-instances-network-instance-interfaces-interface-id`

A unique identifier for this interface - this is expressed as a free-text string

nodetype: leaf (rw)

type: string

If we are passing the name of the intefrace as id=key for a a subinterface, and if there are more than one subinterfaces, then id=key is no more unique. An ideal way to keep it unique would be to provide the name of the subinterface [here *intf.Name + "." + strconv.Itoa(int(id)), or any vendor specific interpretation that uniquely identifies subinterfaces]

Same can be achieved by modifying line `175` as below:
`niIntf := ni.GetOrCreateInterface(*intf.Name + "." + strconv.Itoa(int(id)))`

Reply:
Currently GetOrCreateInterface is using only interface as id. Modified the code to add subinterface to make it unique.

Reply:
Let me rephrase my statement. 
`niIntf := ni.GetOrCreateInterface(name)`
is not equal to
`niIntf.Id := name`, i.e. interface-id that gets configured via gNMI will include subinterface-id regardless 

I.e. while you can do that change, of course, I think that
1) it adds unnecessary complexity (if we decide to add more subinterfaces we'll need to change the code of course, but that would require additional changes anyway)
2) this is not the pattern we use in other tests while using niIntf := ni.GetOrCreateInterface() style of subinterface config

Reply:
> @RanjithansCisco current code will already be using interface.subinterface_id as an interface ID on the wire AFAIK. Give it a try?

@LimeHat the gnmi request does not return sub interface id , it is merely interface id which is passed.

Reply:
> name

@LimeHat first I dont see complexity in modifying just a single line as per the recommendation.
second, please take a look at methods "GetOrCreateInterface" and "NewInterface" in struct-0.go in ondatra.
Path: mod/github.com/openconfig/ondatra@v0.1.5/gnmi/oc/structs-0.go. As per description the id used must be unique.

```
t.Interface[key] = &NetworkInstance_Interface{
    Id: &Id,
} 
```

The argument to GetOrCreateInterface method is assigned to NetworkInstance_Interface.Id

For example, this is already being used here https://github.com/openconfig/featureprofiles/blob/d3acc90dc2f286eff2f82ee20c251629cada2d3a/feature/experimental/policy/policy_vrf_selection/ate_tests/protocol_dscp_rules_for_vrf_selection_test/protocol_dscp_rules_for_vrf_selection_test.go#L208

Reply:
To maintain uniqueness , it is better to have int+subinterface as ID.